contrib/localnet/bitcoin-sidecar/js/src/client.ts

```diff
@@ -0,0 +1,183 @@
+import { initEccLib, payments, Psbt } from "bitcoinjs-lib";
+import { bitcoin, Network, regtest } from "bitcoinjs-lib/src/networks";
+import BIP32Factory, { BIP32Interface } from 'bip32';
+import * as ecc from 'tiny-secp256k1';
+import randomBytes from "randombytes";
+import { ScriptBuilder } from "./script";
+import { Taptree } from "bitcoinjs-lib/src/types";
+import { toXOnly } from "./util";
+
+const LEAF_VERSION_TAPSCRIPT = 0xc0;
+
+initEccLib(ecc);
+const bip32 = BIP32Factory(ecc);
+const rng = randomBytes;
+
+/// The evm address type, a 20 bytes hex string
+export type Address = String;
+export type BtcAddress = String;
+
+/// The BTC transactioin hash returned
+export type BtcTxnHash = String;
+export interface BtcInput {
+    txn: BtcTxnHash,
+    idx: number,
+}
+
+/** 
+ * The example client for interracting with Zetachain in BTC. There are currently two ways 
+ * of calling a smart contract on Zetachain from BTC:
+ * 
+ * - Using OP_RETURN
+ * - Using Witness
+ * 
+ * The method used is now based on the data size. Within 80 bytes, `OP_RETURN` is used, else 
+ * the data is written to Witness.
+ * 
+ * This class handles only the case where data is more than 80 bytes.
+ */
+export class ZetaBtcClient {
+    /** The BTC network interracting with */
+    readonly network: Network;
+
+    private reveal: RevealTxnBuilder | null;
+
+    private constructor(network: Network) {
+        this.network = network;
+    }
+
+    public static regtest(): ZetaBtcClient {
```

Comment:
good to add `testnet()` as well. Features typically goes to ZetaChain testnet and then mainnet.